classpath/java/io/BufferedInputStream.java

```diff
@@ -54,28 +55,7 @@ public int read(byte[] b, int offset, int length) throws IOException {
 
       count += remaining;
       position += remaining;
-      offset += remaining;
-      length -= remaining;
     }
-
-    while (length > 0) {
```

Comment:
OK, what about a compromise? I've written a Test that blocks with avians current BufferedInputStream but runs perfectly well with Oracles JDK. So if you want to keep the loop, lets change it by moving the available() call into the while...
```
    while (length > 0 && in.available() > 0) {
      int c = in.read(b, offset, length);
      if (c == -1) {
        if (count == 0) {
          count = -1;
        }
        break;
      } else {
        offset += c;
        count += c;
        length -= c;
      }
    }
```
This way we still try to fill the given byte array as much as possible, but it will also match the spec and will be compatible to JDK.